gemfirexd/core/src/main/java/com/pivotal/gemfirexd/internal/engine/access/heap/MemHeapScanController.java

```diff
@@ -565,7 +566,7 @@ else if (Misc.getGemFireCache().snapshotEnabled()) {
       // clear the txState so that other thread local is cleared.
       TXManagerImpl.getOrCreateTXContext().clearTXState();
       // gemfire tx shouldn't be cleared in case of row buffer scan
-      if (!(this.getGemFireContainer().isRowBuffer() && lcc.isSkipConstraintChecks())) {
+      if (!(this.getGemFireContainer().isRowBuffer() /*&& lcc.isSkipConstraintChecks()*/)) {
```

Comment:
Earlier used the wrong property from snappy side to differentiate between connection from snappy side and from jdbc. 
Now no support for snapshot on row table so not required. Also realized that this property would have caused no constraint checks for row table.